common/make/js_building_common.mk

```diff
@@ -64,86 +64,71 @@ $(eval $(call CLEAN_RULE,$(JS_BUILD_DIR_ROOT_PATH)))
+# jscomp_off lintChecks: Suppress complaints about code style. TODO: Fix the
+#   code (e.g., switch to modules instead of good.provide) and promote this to
+#   jscomp_error.
+# jscomp_off reportUnknownTypes: Suppressed due to many false positives in case
+#   the compiler isn't smart enough to infer the variable's type.
+# jscomp_off strictMissingProperties: Suppressed due to false positives on the
+#   .base() calls. TODO: Replace the base() calls with the ECMAScript style
+#   inheritance and promote this to jscomp_error.
+# jscomp_off unknownDefines: Suppressed in order to avoid compiler complaints
+#   when an unused constant is passed via --define.
 JS_BUILD_COMPILATION_FLAGS += \
 	--compilation_level=SIMPLE \
 	--define='GoogleSmartCard.ExecutableModule.TOOLCHAIN=$(TOOLCHAIN)' \
 	--define='GoogleSmartCard.Logging.USE_SCOPED_LOGGERS=false' \
-	--dependency_mode=STRICT \
-	--jscomp_error=accessControls \
-	--jscomp_error=ambiguousFunctionDecl \
-	--jscomp_error=checkRegExp \
-	--jscomp_error=checkTypes \
-	--jscomp_error=checkVars \
-	--jscomp_error=conformanceViolations \
-	--jscomp_error=const \
-	--jscomp_error=constantProperty \
-	--jscomp_error=deprecated \
-	--jscomp_error=deprecatedAnnotations \
-	--jscomp_error=duplicateMessage \
-	--jscomp_error=es3 \
-	--jscomp_error=es5Strict \
-	--jscomp_error=externsValidation \
-	--jscomp_error=fileoverviewTags \
-	--jscomp_error=functionParams \
-	--jscomp_error=globalThis \
-	--jscomp_error=invalidCasts \
-	--jscomp_error=lintChecks \
-	--jscomp_error=misplacedTypeAnnotation \
-	--jscomp_error=missingGetCssName \
-	--jscomp_error=missingOverride \
-	--jscomp_error=missingPolyfill \
-	--jscomp_error=missingProperties \
-	--jscomp_error=missingProvide \
-	--jscomp_error=missingRequire \
-	--jscomp_error=missingReturn \
-	--jscomp_error=missingSourcesWarnings \
-	--jscomp_error=msgDescriptions \
-	--jscomp_error=newCheckTypes \
-	--jscomp_error=nonStandardJsDocs \
-	--jscomp_error=suspiciousCode \
-	--jscomp_error=strictModuleDepCheck \
-	--jscomp_error=typeInvalidation \
-	--jscomp_error=undefinedNames \
-	--jscomp_error=undefinedVars \
-	--jscomp_error=underscore \
-	--jscomp_error=unusedLocalVariables \
-	--jscomp_error=unusedPrivateMembers \
-	--jscomp_error=uselessCode \
-	--jscomp_error=useOfGoogBase \
-	--jscomp_error=visibility \
+	--dependency_mode=PRUNE \
+	--jscomp_error='*' \
+	--jscomp_off extraRequire \
+	--jscomp_off jsdocMissingConst \
+	--jscomp_off lintChecks \
 	--jscomp_off reportUnknownTypes \
+	--jscomp_off strictMissingProperties \
 	--jscomp_off unknownDefines \
 	--use_types_for_optimization=false \
 	--warning_level=VERBOSE \
 
 
-#
 # Closure Compiler compilation flags that should be additionally specified when
 # compiling JavaScript files with tests.
 #
-
+# Explanation:
+# dependency_mode=PRUNE_LEGACY: In addition to the standard PRUNE behavior (see
+#   above), also include all JS files that don't have a good.provide. This
```

Comment:
typo: goog*

Reply:
Done, thanks for catching this! I'm making this typo all the time - probably my hidden excitement of Closure Compiler :)